src/simple_http.rs

```diff
@@ -230,8 +230,16 @@ impl Builder {
         };
         // (2) split off path
         let (before_path, path) = {
-            let mut split = after_scheme.splitn(2, "/");
-            (split.next().unwrap(), format!("/{}", split.next().unwrap_or("")))
+            if let Some(slash) = after_scheme.find("/") {
+                (
+                    &after_scheme[0..slash],
+                    // Replace double slashes that might occur;
+                    // this is pretty common when adding paths together.
+                    after_scheme[slash..].replace("//", "/")
+                )
+            } else {
+                (after_scheme, "/".to_owned())
+            }
```

Comment:
the url is already checked for doubled-`/` just after the host part. However i'm not sure it's a good idea to sanitize the caller's input: it's reasonable to expect they are giving a decent path, and we might not do the right thing. For example on Windows, which does some nasty stuff with paths.

Reply:
Hmm, I agree messing with user input is tricky. Though I think double slashes are valid in paths no? In any way, if you give url `http://localhost:1234//wallet/name`, bitcoind errors. But yeah that might be just something we shouldn't support.

Reply:
> In any way, if you give url http://localhost:1234//wallet/name, bitcoind errors

I thought
```rust
            let mut split = after_scheme.splitn(2, "/");
            (split.next().unwrap(), format!("/{}", split.next().unwrap_or("")))
```
prevented that but it doesn't if the user provides a `/path/to/wallet` so i now suggest to either silently correct the first character (drop it if it's a `/` already), or document in `path` that we want an "absolute" path or not. The former seems preferable